public/app/core/components/FormPrompt/Prompt.tsx

```diff
@@ -0,0 +1,27 @@
+import * as H from 'history';
+import { useEffect } from 'react';
+
+import { locationService } from '@grafana/runtime';
+
+interface PromptProps {
+  when?: boolean;
+  message: string | ((location: H.Location) => string | boolean);
+}
+
+export const Prompt = ({ message, when = true }: PromptProps) => {
+  const history = locationService.getHistory();
+
+  useEffect(() => {
+    if (!when) {
+      return undefined;
+    }
+    //@ts-expect-error TODO Update the history package to fix types
```

Comment:
Do we have an issue for this? I didn't find one in the [epic](https://github.com/grafana/grafana/issues/83666).

Reply:
Good point, I'll need to add it, altho I suspect that might be an epic on its own.